anta/tests/bfd.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2023-2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+"""
+BFD test functions
+"""
+# Mypy does not understand AntaTest.Input typing
+# mypy: disable-error-code=attr-defined
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Optional
+
+from anta.models import AntaCommand, AntaTest
+
+
+class VerifyBFDPeersHealth(AntaTest):
+    """
+    Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF.
+    Expected results:
+        * success: The test will pass if IPv4 BFD peers are not down, remote disc is not zero and last down above the defined threshold for all VRF.
+        * failure: The test will fail if IPv4 BFD peers are down, remote disc is zero and last down below the defined threshold for all VRF.
+    """
+
+    name = "VerifyBFDPeersHealth"
+    description = "Verifies there is no IPv4 BFD peer in the down state, remote disc is not zero and last down should be above the threshold for all VRF."
+    categories = ["bfd"]
+
+    # revision 1 as later revision introduces additional nesting for type
+    commands = [AntaCommand(command="show bfd peers", revision=1), AntaCommand(command="show clock")]
+
+    class Input(AntaTest.Input):
+        """
+        This class defines the input parameters of the testcase.
+        """
+
+        last_down: Optional[int] = None
```

Comment:
```suggestion
        last_down: Optional[int] = Field(default=None, gt=0)
```

Reply:
Also, I am not sure I like `last_down` as an input. It is not very descriptive.

Reply:
i can update the doc string as 
`Optional last down threshold in hours to check if a bfd peer was down before those hours or not.`

Reply:
Maybe `down_threshold`?